crates/goose/src/providers/deepseek.rs

```diff
@@ -0,0 +1,133 @@
+use anyhow::Result;
+use async_trait::async_trait;
+use reqwest::Client;
+use serde_json::Value;
+use std::time::Duration;
+
+use super::base::{ConfigKey, Provider, ProviderMetadata, ProviderUsage};
+use super::errors::ProviderError;
+use super::utils::{emit_debug_trace, get_model, handle_response_openai_compat};
+use crate::message::Message;
+use crate::model::ModelConfig;
+use crate::providers::formats::openai::{create_request, get_usage, response_to_message};
+use mcp_core::tool::Tool;
+
+pub const DEEPSEEK_DEFAULT_MODEL: &str = "deepseek-chat";
+
+// Deepseek can run two models, the default is recommended
+pub const DEEPSEEK_KNOWN_MODELS: &[&str] = &[
+    "deepseek-chat",
```

Comment:
curious if you tested how well this works? the website says function calling is unstable. 
https://api-docs.deepseek.com/guides/function_calling